x-pack/plugin/eql/qa/common/src/main/resources/test_sample.toml

```diff
@@ -0,0 +1,252 @@
+[[queries]]
+name = "anyRulesWithOneDifferentNameKey"
+query = '''
+  sample by host
+    [any where uptime > 0]   by os
+    [any where port > 100]   by op_sys 
+    [any where bool == true] by os
+'''
+expected_event_ids  = [17,26,16,
```

Comment:
hey @astefan! it's been a while! 🙂
cool feature, looks similar to Endgame EQL `join` but without all the baggage that word brings, so I'm glad to see the name change.

I think it behaves how I would intuitively expect, just want to double check one property: how are the resulting events ordered? Is it (1) in declaration order, parallel to the structure of the query? Or is it (2) chronological order, not mirroring the structure.

I can understand arguments for both, and I think (1) makes the most intuitive sense to me and that was how the Endgame `join` syntax behaved. From a quick glance, it looks like this is the current behavior, which sounds like the right call.

Glad to see this happen, it's exciting!

Reply:
excellent! that's what it looked like. nice work with the feature!